src/DeviceListener.ts

```diff
@@ -66,7 +65,7 @@ export default class DeviceListener {
     private ourDeviceIdsAtStart: Set<string> | null = null;
     // The set of device IDs we're currently displaying toasts for
     private displayingToastsForDeviceIds = new Set<string>();
-    private running = false;
+    private runningClient?: MatrixClient;
```

Comment:
tbh it feels a bit odd to conflate `running` and `client`: personally I would keep them separate, with a comment on `client` along the lines of "only defined when 'running' is true", even if, strictly speaking, `running` is then redundant.